File: Core/Models/Coach.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Models
{
    public sealed class Coach : _ParentPerson
    {
        public string WhatTrains { get; set; }
        [ForeignKey("CoachClub")]
        public Guid CoachClubId { get; set; }
        public SportsClub? CoachClub { get; set; }
        private Coach():base() { }
        public Coach(string coachRole, DateTime contractFrom, DateTime contractTo, string firstName, string lastName):base(contractFrom, contractTo, firstName, lastName) 
        {
            if (string.IsNullOrEmpty(coachRole))
                throw new Exception("CoachRole_Cannot_Be_Null_CoachCtor");
            WhatTrains = coachRole;
        }
        public void Update(Coach model)
        {
            this.FirstName = model.FirstName;
            this.LastName = model.LastName;
            this.ContractFrom = model.ContractFrom;
            this.ContractTo = model.ContractTo;
            this.WhatTrains = model.WhatTrains;
        }
    }
}

File: FrontEnd_Angular/src/app/shared/Interfaces/Coach.ts
export interface Coach {
  Id: string;
  Status: number;
  CreateAt: Date;
  UpdateAt: Date;
  ContractFrom: Date;
  ContractTo: Date;
  FirstName: string;
  LastName: string;

  WhatTrains: string;
  CoachClubId: string;
}
export interface CoachDTO {
  id: string;
  contractFrom: Date;
  contractTo: Date;
  firstName: string;
  lastName: string;
  whatTrains: string;
}
export interface CoachAddCommand {
  ClubId: string;
  ContractFrom: Date;
  ContractTo: Date;
  FirstName: string;
  LastName: string;
  WhatTrains: string;
}
export interface CoachEditCommand {
  CoachId: string;
  ContractFrom: Date;
  ContractTo: Date;
  FirstName: string;
  LastName: string;
  WhatTrains: string;
}
